### DISCUSSION TITLE
Trouble detecting/decoding weather station

### DISCUSSION THREAD

**Discussion by patienttruth** [merbanan/rtl_433#3115] [Q&A] [answered]
Greetings,

I've spent most of the day on this, but no success. My goal is to pass my Acurite weather station data to Home Assistant.

I have a RTL-SDR v4 with a cheap [antenna](https://www.amazon.com/dp/B094MW1YMV). I'm hoping to use it w/Home Assistant, however I'm not getting to the point of receiving a signal I can pass on. I'm wondering if either I have bad hardware, or antenna configuration.

I was thinking I could place the antenna in my office since we have the Acurite display in the kitchen and it works fine. (So there must be a signal?) 

Currently the SDR is plugged into a 6' USB cable, and into my laptop's powered hub. The antenna is on a file cabinet 6' away, and there is a single wall to the outside. The wall is drywall, fiberglass insulation, plywood, hardiboard, just like the kitchen, except there are multiple walls. The weather station is about 50' total from the antenna.

Any guidance to troubleshoot my hardware/config would be appreciated.

Here are the outputs of `rtl_433 -S all`:
<details>
<summary>Terminal Output</summary>

```
$ rtl_433 -S all
rtl_433 version 24.10-15-g54837576 branch master at 202412132021 inputs file rtl_tcp RTL-SDR
Found Rafael Micro R828D tuner
[SDR] Using device 0: RTLSDRBlog, Blog V4, SN: 00000001, "Generic RTL2832U OEM"
Exact sample rate is: 250000.000414 Hz
[R82XX] PLL not locked!
[R82XX] PLL not locked!
Allocating 15 zero-copy buffers
*** Saving signal to file g001_433.92M_250k.cu8 (635382 samples, 1310720 bytes)
bitbuffer_add_bit: Warning: row count limit (50 rows) reached
bitbuffer_add_bit: Warning: row count limit (50 rows) reached
bitbuffer_add_bit: Warning: row count limit (50 rows) reached
bitbuffer_add_bit: Warning: row count limit (50 rows) reached
bitbuffer_add_bit: Warning: row count limit (50 rows) reached
bitbuffer_add_bit: Warning: row count limit (50 rows) reached
Signal bigger than buffer, signal = 10354688 > buffer 3145728 !!
*** Saving signal to file g002_433.92M_250k.cu8 (5173065 samples, 3145728 bytes)
bitbuffer_add_bit: Warning: row count limit (50 rows) reached
bitbuffer_add_bit: Warning: row count limit (50 rows) reached
bitbuffer_add_bit: Warning: row count limit (50 rows) reached
bitbuffer_add_bit: Warning: row count limit (50 rows) reached
bitbuffer_add_bit: Warning: row count limit (50 rows) reached
bitbuffer_add_bit: Warning: row count limit (50 rows) reached
Signal bigger than buffer, signal = 11796480 > buffer 3145728 !!
*** Saving signal to file g003_433.92M_250k.cu8 (5877040 samples, 3145728 bytes)
bitbuffer_add_bit: Warning: row count limit (50 rows) reached
bitbuffer_add_bit: Warning: row count limit (50 rows) reached
Signal bigger than buffer, signal = 3670016 > buffer 3145728 !!
*** Saving signal to file g004_433.92M_250k.cu8 (1812160 samples, 3145728 bytes)
*** Saving signal to file g005_433.92M_250k.cu8 (1288450 samples, 2621440 bytes)
bitbuffer_add_bit: Warning: row count limit (50 rows) reached
bitbuffer_add_bit: Warning: row count limit (50 rows) reached
bitbuffer_add_bit: Warning: row count limit (50 rows) reached
Signal bigger than buffer, signal = 5505024 > buffer 3145728 !!
*** Saving signal to file g006_433.92M_250k.cu8 (2702821 samples, 3145728 bytes)
bitbuffer_add_bit: Warning: row count limit (50 rows) reached
bitbuffer_add_bit: Warning: row count limit (50 rows) reached
bitbuffer_add_bit: Warning: row count limit (50 rows) reached
Signal bigger than buffer, signal = 7995392 > buffer 3145728 !!
*** Saving signal to file g007_433.92M_250k.cu8 (3941781 samples, 3145728 bytes)
bitbuffer_add_bit: Warning: row count limit (50 rows) reached
bitbuffer_add_bit: Warning: row count limit (50 rows) reached
bitbuffer_add_bit: Warning: row count limit (50 rows) reached
bitbuffer_add_bit: Warning: row count limit (50 rows) reached
bitbuffer_add_bit: Warning: row count limit (50 rows) reached
bitbuffer_add_bit: Warning: row count limit (50 rows) reached
Signal bigger than buffer, signal = 10354688 > buffer 3145728 !!
*** Saving signal to file g008_433.92M_250k.cu8 (5163151 samples, 3145728 bytes)
bitbuffer_add_bit: Warning: row count limit (50 rows) reached
*** Saving signal to file g009_433.92M_250k.cu8 (1168534 samples, 2359296 bytes)
bitbuffer_add_bit: Warning: row count limit (50 rows) reached
bitbuffer_add_bit: Warning: row count limit (50 rows) reached
Signal bigger than buffer, signal = 4063232 > buffer 3145728 !!
*** Saving signal to file g010_433.92M_250k.cu8 (1986517 samples, 3145728 bytes)
bitbuffer_add_bit: Warning: row count limit (50 rows) reached
*** Saving signal to file g011_433.92M_250k.cu8 (1151578 samples, 2359296 bytes)
bitbuffer_add_bit: Warning: row count limit (50 rows) reached
*** Saving signal to file g012_433.92M_250k.cu8 (1160324 samples, 2359296 bytes)
bitbuffer_add_bit: Warning: row count limit (50 rows) reached
bitbuffer_add_bit: Warning: row count limit (50 rows) reached
Signal bigger than buffer, signal = 4325376 > buffer 3145728 !!
*** Saving signal to file g013_433.92M_250k.cu8 (2162193 samples, 3145728 bytes)
bitbuffer_add_bit: Warning: row count limit (50 rows) reached
*** Saving signal to file g014_433.92M_250k.cu8 (865045 samples, 1835008 bytes)
*** Saving signal to file g015_433.92M_250k.cu8 (644914 samples, 1310720 bytes)
*** Saving signal to file g016_433.92M_250k.cu8 (35280 samples, 131072 bytes)
*** Saving signal to file g017_433.92M_250k.cu8 (35279 samples, 131072 bytes)
^CSignal caught, exiting!
```

</details>

[2024-12-14 - rtl_433.zip](https://github.com/user-attachments/files/18138601/2024-12-14.-.rtl_433.zip)
[2024-12-14 - rtl_433-2.zip](https://github.com/user-attachments/files/18138602/2024-12-14.-.rtl_433-2.zip)

**Comment by zuckschwerdt** [comment, marked as answer]
Inspect the files in https://triq.org/pdv/ -- they are empty with high noise level, that's why rtl_433 triggers and overflows with random data.
Try `rtl_433 -Y autolevel -M level -M noise` to watch signal levels, not sure what to try though. Check if you have the Blog-V4 enabled librtlsdr?

**Reply by patienttruth** [reply]
Good morning,

Thank you! 

I noticed the readings looked bright compared to some of the screenshots I was seeing. I do have a Z-Wave and ZigBee controller like 10-20' from where I had the antenna, but they're on quite different frequency so that shouldn't be an issue right?

I'll run those options to save and post the info back, but when I tried it yesterday around -13 was the highest, with most reading -6 or -8, so I figured that wasn't the issue. (I'm not a radio expert, clearly, just going off many different help theeads

**Reply by zuckschwerdt** [reply]
Good values would be around -20 dB for the noise level, past -15 will be trouble and closer to -0 dB is a jammed band. rtl_433 needs at least 9 dB SNR (signal to noise ratio) and will always assume a signal above -12 dB to be data.

**Reply by patienttruth** [reply]
> Good values would be around -20 dB for the noise level, past -15 will be trouble and closer to -0 dB is a jammed band. rtl_433 needs at least 9 dB SNR (signal to noise ratio) and will always assume a signal above -12 dB to be data.

That's very helpful. Thank you.

So, progress from your tip above!

I originally installed with these [instructions](https://blog.fuzzymistborn.com/post-dark-sky-weather-part-2/) which must have resulted in the wrong drivers.

This morning I moved the SDR to my office pc so I could troubleshoot from the warmth of my house (point being, fresh install). For the fresh install I used this guide from [RTL-SDR Blog (V4 User's Guide)](https://www.rtl-sdr.com/v4/). Then I installed rtl-433 with `apt-get install rtl-433` per the readme here.

now, `rtl_433` works and my weather station comes up immediately. So know I know the hardware is good. Thank you @zuckschwerdt!

My next steps are to try to change the drivers on my server laptop and thy `rtl_433` over there again. If good I'll try to pass to Home Assistant.

Thank you @zuckschwerdt!

**Reply by zuckschwerdt** [reply]
The rtl_433 version from apt will be old, but if the station you want to receive is already supported then there is no need to compile yourself. If you do advanced things you likely want a current version at some point ;)

We also provide binaries https://github.com/merbanan/rtl_433/releases -- you just need to pick the right OpenSSL version (no harm to try each) and cpu type.

Like you read in the RTL-SDR Blog V4 Users Guide, the librtlsdr driver needs an update for the V4 currently (it might be included in the next Debian release).
